### PR TITLE
AfterEffects: add review flag to each instance

### PR DESCRIPTION
**Pull request by kalisp**
## Changelog Description
Adds `mark_for_review` flag to the Creator to allow artists to disable review if necessary.

Exposed this flag in Settings, by default set to True (eg. same behavior as previously).

## Additional info
Not sure if `CollectReview` should be separate plugin and exposed in Settings. I matched state from Photoshop to be the same.
Adding 'review' family from `creator_attributes` could be easily moved to `collect_render.py` and whole `collect_review.py` removed.

## Testing notes:
1. toggle `Review` flag in created `render` instance in AE 
2. publish locally
3. publish on Deadline

### CODE REVIEW

**Comment by 64qam** [comment]
Review works fine on local publishing, publishing on Deadline ignores if the review is on or off and always creates review.

File: openpype/hosts/aftereffects/plugins/publish/collect_review.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Provides:
+    instance     -> family ("review")
+"""
+
+import os
+
+import pyblish.api
+
+from openpype.pipeline.create import get_subset_name
```
Comment (iLLiCiTiT) [inline openpype/hosts/aftereffects/plugins/publish/collect_review.py:13]:
```suggestion
```

File: openpype/hosts/aftereffects/plugins/publish/collect_review.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class CollectReview(pyblish.api.InstancePlugin):
+    """Add review to families if instance created with 'mark_for_review' flag
+    """
+
+    label = "Collect Review"
+    label = "Review"
+    hosts = ["aftereffects"]
+    order = pyblish.api.CollectorOrder + 0.1
+
+    publish = True
```
Comment (iLLiCiTiT) [inline openpype/hosts/aftereffects/plugins/publish/collect_review.py:25]:
```suggestion
```

File: openpype/hosts/aftereffects/plugins/publish/collect_review.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    def process(self, instance):
+        creator_attributes = instance.data.get("creator_attributes")
+        if (creator_attributes and
+                creator_attributes.get("mark_for_review") and
+                "review" not in instance.data["families"]):
+            instance.data["families"].append("review")
```
Comment (iLLiCiTiT) [inline openpype/hosts/aftereffects/plugins/publish/collect_review.py:32]: Unused attributes `publish`, doubled `label` and the plugin does not have any instance filtering so can be changed to context plugin (faster processing).
```suggestion
class CollectReview(pyblish.api.ContextPlugin):
    """Add review to families if instance created with 'mark_for_review' flag
    """

    label = "Review"
    hosts = ["aftereffects"]
    order = pyblish.api.CollectorOrder + 0.1

    def process(self, context):
        for instance in context:
            creator_attributes = instance.data.get("creator_attributes") or {}
            if (
                creator_attributes.get("mark_for_review")
                and "review" not in instance.data["families"]
            ):
                instance.data["families"].append("review")
```

File: openpype/hosts/aftereffects/plugins/publish/collect_render.py [hunk 4]
```diff
@@ -139,6 +139,10 @@ def get_instances(self, context):
                 instance.toBeRenderedOn = "deadline"
                 instance.renderer = "aerender"
                 instance.farm = True  # to skip integrate
+                if "review" in instance.families:
+                    # to skip ExtractReview locally
```
Comment (iLLiCiTiT) [inline openpype/hosts/aftereffects/plugins/publish/collect_render.py:143]: But, aren't you changing the farm instance here?

File: openpype/hosts/aftereffects/plugins/publish/collect_review.py (hunk 1 above)
Comment (iLLiCiTiT) [inline openpype/hosts/aftereffects/plugins/publish/collect_review.py:13]:
```suggestion
import pyblish.api
```

**Pull request merged**